pkg/network/common/dns.go

```diff
@@ -84,14 +99,23 @@ func (d *DNS) Get(dns string) dnsValue {
 }
 
 func (d *DNS) Add(dns string) error {
+	// This is a blocking operation, therefore must be done before acquring
+	// the lock
+	ips, ttl, err := d.getIPsAndMinTTL(dns)
+	if err != nil {
+		return err
+	}
+
+	defer func(start time.Time) {
+		warnSlowOperation("Add resolved egress DNS record", start)
```

Comment:
shouldn't the defer func be declared before calling getIPsAndMinTTL() if you want to warn of slow operation when making the dns query?

Reply:
What I wanted to log was precisely the bits that we can control and optimize (if the DNS is slow, we can't do anything about it).

Said so I agree with logging slow DNS queries is useful as well, but I'm not quite sure what's an appropriate threshold for that. I tried doing a few DNS queries to 8.8.8.8 and 1.1.1.1 and I'm getting somewhere between 20 and 80 ms most of the time. Do 300 milliseconds sound like a reasonable